lib/elementSearch.js

```diff
@@ -95,46 +95,54 @@ async function matchContainsElements(
       tagName,
       textToTranslate,
       translateTo,
-      xpathText,
+      xpathText,options,
     );
   if (tagName === 'button' && (!elements || !elements.length))
     elements = await containsTextInChildElements(
       tagName,
       textToTranslate,
       translateTo,
-      xpathText,
+      xpathText,options,
     )
   return elements;
 }
 
 async function matchTextNode(
   textToTranslate,
   translateTo,
-  xpathText,
+  xpathText,options,
 ) {
-  return await $$xpath(
-    `//text()[translate(normalize-space(string()), "${textToTranslate}", "${translateTo}")=${xpathText}]`,
+  return options ? await $$xpath(
+    `//text()[translate(normalize-space(string()), "${textToTranslate}", "${translateTo}")=${xpathText}]`,options.selectHiddenElement
+  ) : await $$xpath(
+      `//text()[translate(normalize-space(string()), "${textToTranslate}", "${translateTo}")=${xpathText}]`,
   );
```

Comment:
We don't need this check here as [$$xpath](https://github.com/getgauge/taiko/blob/73243dee5cbddb9e69ae30b4391b8d26a288f20e/lib/elementSearch.js#L249) already takes care of this scenario. I.M.O we should just pass the `selectHiddenElement` option to the `$$xpath` function.

Reply:
@negiDharmendra thanks, i have tried without this condition but getting exception, thats why  handled it